src/providers/cortex/trulens/providers/cortex/provider.py

```diff
@@ -110,47 +116,24 @@ def __init__(
             )
         super().__init__(**self_kwargs)
 
-    def _exec_snowsql_complete_command(
+    def _invoke_cortex_complete(
         self,
         model: str,
         temperature: float,
         messages: Optional[Sequence[Dict]] = None,
-    ):
+    ) -> str:
         # Ensure messages are formatted as a JSON array string
         if messages is None:
             messages = []
 
-        messages_json_str = json.dumps(messages)
-
         options = {"temperature": temperature}
 
-        options_json_str = json.dumps(options)
-
-        completion_input_str = """
-            SELECT SNOWFLAKE.CORTEX.COMPLETE(
-                ?,
-                parse_json(?),
-                parse_json(?)
-            )
-        """
-        if (
-            hasattr(self.snowflake_conn, "_paramstyle")
-            and self.snowflake_conn._paramstyle == "pyformat"
-        ):
-            completion_input_str = completion_input_str.replace("?", "%s")
-
-        # Executing Snow SQL command requires an active snow session
-        cursor = self.snowflake_conn.cursor()
-        try:
-            cursor.execute(
-                completion_input_str,
-                (model, messages_json_str, options_json_str),
-            )
-            result = cursor.fetchall()
-        finally:
-            cursor.close()
-
-        return result
+        completion_res_str: str = Complete(
```

Comment:
1. How is the snowpark session inferred?
2. Can you check it's not using snowpark_session.sql? As before I'm worried about the thread-safety.

Reply:
1. will update to explicitly take in snow session 
2. Yes I'm very certain it doesn't use the sql function, only REST api (see my other comment)